### PR TITLE
Ipc mailbox calls export

### PR DESCRIPTION
**Pull request by SamuelVLC**
- [x] Export of IPC system calls ff682635985a6ebd980b37642e3e94df50988754
- [ ] IPC system calls testing b1e65069c9fe939bcd3057216c24d2c21278d8bf

### CODE REVIEW

File: include/nanvix/kernel/kcall.h [hunk 1]
```diff
@@ -31,57 +31,65 @@
...
+#define NR_spawn 17               /** kernel_spawn()            */
+#define NR_semget 18              /** kernel_semget()           */
+#define NR_semop 19               /** kernel_semop()            */
+#define NR_semctl 20              /** kernel_semctl()           */
+#define NR_thread_get_id 21       /** kernel_thread_get_id()    */
+#define NR_thread_create 22       /** kernel_thread_create()    */
+#define NR_thread_exit 23         /** kernel_thread_exit()      */
+#define NR_thread_yield 24        /** kernel_thread_yield()     */
+#define NR_mailbox_tag 25         /** kernel_mailbox_tag        */
+#define NR_mailbox_is_assigned 26 /** kernel_mailbox_is_assigned()*/
```
Comment (ppenna) [inline include/nanvix/kernel/kcall.h:60]: Do we really need a kernel call for this? Taka some inspiration on the file system calls available on Linux/Windows. Do we have an equivalent function?

File: src/servers/test/src/test.rs [hunk 2]
```diff
@@ -439,14 +439,119 @@ fn test_semget_call() -> bool {
...
+        }    
+    }
+
+    true
+}
+
+// Test systemcall for method of ipc module
+fn test_mailbox_owner() -> bool {
+    let mut owner: i32 = 0;
+    for mbxid in 0..256 {
```
Comment (ppenna) [inline src/servers/test/src/test.rs:467]: Could we replace 256 by some constant?

File: src/servers/test/src/test.rs [hunk 3]
```diff
@@ -501,5 +606,10 @@ pub fn test_kernel_calls() {
     test!(test_semget_call());
     test!(test_semop_call());
     test!(test_semctl_call());
-    test!(test_mailbox_tag());
+    test!(test_mailbox_modules());
+//  test!(test_mailbox_default()); //Test yet to be defined
```
Comment (ppenna) [inline src/servers/test/src/test.rs:610]: Usually you can use the `unimplemented!()` macro for that. Not sure if it is available out of the std crate, but it is worth trying.

Comment (ppenna) [reply src/servers/test/src/test.rs:610]: I've just realized this belongs to `std`. Let's leave this to later.

File: src/servers/test/src/test.rs (hunk 2 above)
Comment (SamuelVLC) [reply src/servers/test/src/test.rs]: solved in #https://github.com/nanvix/microkernel/pull/408/commits/3a4b0d390a7a8a7f24c904935618bb40e93a3ec4

File: include/nanvix/kernel/kcall.h (hunk 1 above)
Comment (SamuelVLC) [reply include/nanvix/kernel/kcall.h]: system calls removed in https://github.com/nanvix/microkernel/pull/408/commits/3a4b0d390a7a8a7f24c904935618bb40e93a3ec4

File: src/kernel/kcall/mod.h [hunk 4]
```diff
@@ -225,6 +225,82 @@ extern int kcall_semget(unsigned key);
...
 
+/**
+ * @brief Checks whether a mailbox is assigned.
+ *
+ * @param mbxid mailbox ID.
+ *
+ * @returns Upon successful (mailbox is assigned), returns 1. Upon Failure,
+ * returns 0.
+ */
+extern int kcall_mailbox_is_assigned(int mbxid);
```
Comment (ppenna) [inline src/kernel/kcall/mod.h:236]: Should we remove this one?

**Pull request closed without merging**